go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -411,7 +412,10 @@ func (npw *nodePortWatcher) updateServiceInfo(index ktypes.NamespacedName, servi
 func addServiceRules(service *kapi.Service, svcHasLocalHostNetEndPnt bool, npw *nodePortWatcher) {
 	// For dpu or Full mode
 	if npw != nil {
-		npw.updateServiceFlowCache(service, true, svcHasLocalHostNetEndPnt)
```

Comment:
hmm.... Can we percolate this error back to the caller as well? It should ensure that a retry will make sure this works as expected.
I know that this needs to be done to all places that invoke this as well, but we are best not to just log the error anymore.

Reply:
> retry only for specific transient errors ?

I think we will retry up to a limit (60 secs?) on sync and then do a fatal panic which will have the pod restarted.
We currently consider all errors to be retriable. That is why/how we need to make sure the functions are idem-potent.

Reply:
yes, for sync services, retries happens for 60s in case of failure and on top of this waitTask retries for 5 mins before  ovnkube node pod startup to fail.
but in case of service events, error can't be returned, so ovnkube node itself can retry for 60s with proposed changes.
these errors are mostly thrown only for invalid ingress IP or non existent IP table or chain. so i think it's safe to make a retry.
afaiu these methods are already idempotent as it always trying to overwrite service flow cache and iptables for a given service.